pkg/provider/kubernetes/crd/kubernetes.go

```diff
@@ -509,6 +510,49 @@ func buildTLSOptions(ctx context.Context, client Client) map[string]tls.Options
 	return tlsOptions
 }
 
+func buildTLSStores(ctx context.Context, client Client) map[string]tls.Store {
+	tlsStoreCRD := client.GetTLSStores()
+	var tlsStores map[string]tls.Store
+
+	if len(tlsStoreCRD) == 0 {
+		return tlsStores
+	}
+	tlsStores = make(map[string]tls.Store)
+
+	for _, tlsStore := range tlsStoreCRD {
+		namespace := tlsStore.Namespace
+		logger := log.FromContext(log.With(ctx, log.Str("tlsStore", tlsStore.Name), log.Str("namespace", namespace)))
+		secretName := tlsStore.Spec.DefaultCertificate.SecretName
+
+		secret, exists, err := client.GetSecret(namespace, secretName)
+		if err != nil {
+			logger.Errorf("Failed to fetch secret %s/%s: %v", namespace, secretName, err)
+			return tlsStores
```

Comment:
I'm not so sure about this (and the others below) return.
If we are in "failure mode", shouldn't we rather return nil, instead of a half-processed tlsStores? otoh, if we don't consider this error as critical, shouldn't we keep going through the stores? (with a continue statement)

Reply:
I think that the continue would be better, as it would be non-blocking, and would match the failure process of TLS Options.